test/functional/tests/fault_injection/test_error_device_as_cache.py

```diff
@@ -0,0 +1,260 @@
+          errors are present in Open CAS stats.
+        pass_criteria:
+          - I/O error count in fio are zero
+          - I/O error count in cache statistics are greater than I/O error count in fio
+    """
+    with TestRun.step("Prepare error device and setup cache and core."):
+        cache, cores, error_dev, cache_dev = prepare_configuration(cache_mode, cache_line_size, 1)
+
+    if cache_mode == CacheMode.WA:
+        with TestRun.step("Read core to null block."):
+            Dd().input(cores[0]) \
+                .output('/dev/null') \
+                .iflag('direct') \
+                .block_size(cache_line_size.value) \
+                .run()
+
+    with TestRun.step("Run fio against core to fill it with pattern."):
+        fio = (
+            Fio()
+            .create_command()
+            .target(cores[0])
+            .io_engine(IoEngine.libaio)
+            .io_depth(1)
+            .num_jobs(1)
+            .size(cache_part_size)
+            .read_write(ReadWrite.randwrite)
+            .block_size(cache_line_size)
+            .direct()
+            .verification_with_pattern("0xabcd")
+            .do_verify(False)
+            .continue_on_error(ErrorFilter.io)
+        )
+        fio_errors = fio.run()[0].total_errors()
+
+    with TestRun.step("Check if fio reported no write errors."):
+        if fio_errors != 0:
+            TestRun.fail("Fio reported write errors!")
+
+    with TestRun.step("Stop cache and enable errors on cache device (after metadata area)."):
+        cache_size = cache.size
+        metadata_size = cache.get_metadata_size() + Size(1, Unit.MiB)
+        cache.stop()
+
+        error_dev.change_table(
+            error_table(cache_size, metadata_size).fill_gaps(cache_dev)
+        )
+
+    with TestRun.step("Load cache and reset counters."):
+        cache = casadm.load_cache(error_dev)
+        cache.reset_counters()
+
+    with TestRun.step("Run fio against core with pattern verification."):
+        fio = (
+            Fio()
+            .create_command()
+            .target(cores[0])
+            .io_engine(IoEngine.libaio)
+            .io_depth(1)
+            .num_jobs(1)
+            .size(cache_part_size)
+            .read_write(ReadWrite.randread)
+            .block_size(cache_line_size)
+            .direct()
+            .verification_with_pattern("0xabcd")
+            .do_verify(False)
+            .continue_on_error(ErrorFilter.io)
+        )
+        fio_errors = fio.run()[0].total_errors()
+
+    with TestRun.step("Check if fio reported no read errors."):
+        if fio_errors != 0:
+            TestRun.fail("Fio reported read errors!")
+
+    with TestRun.step("Check cache error statistics."):
+        stats = cache.get_statistics()
+        read_errors_in_cache = stats.error_stats.cache.reads
+        total_errors_in_cache = stats.error_stats.cache.total
+        if read_errors_in_cache < fio_errors:
```

Comment:
do you know why we check "<" instead of "!=" ?

Reply:
Actually, after some sleep, I think the checks in these two tests can be done in a different way. I will redesign some parts of both tests.